### PR TITLE
Cert 2

### PR DESCRIPTION
**Pull request by Jatewo**
## Description

Fixed errors related to login and made functionality for creating and checking login tokens.

## Changes

- Removed setPassword("") from login function
- Added functionality for setting login token upon succesful login
- Checks for login
  - Login page redirects to home page if token is already present (user is already logged in)
  - All other pages redirects to login page if no token is present (user is not logged in)
- Removed unused leftover code
- Added component for popup alerts
- Added helper function for removing emojis from password input
- Added component for alerts in form fields
- Refactored login and register forms to use the alert component
- Fixed error handling when logging in (shows error for wrong email and password)
- Relevant tests for login and register screens and components

#### #25 - Dependency on Authentication:
Token and user info are saved in AsyncStorage. User info might need changes in the backend in order to send the needed information.

## Related Backlog Items

- Certificate Backlog [Item 1 - Register](https://github.com/orgs/Educado-App/projects/5?pane=issue&itemId=38249106)
- Certificate Backlog [Item 2 - Logout](https://github.com/orgs/Educado-App/projects/5/views/1?pane=issue&itemId=38249145)
- Certificate Backlog [Item 4 - Login](https://github.com/orgs/Educado-App/projects/5/views/1?pane=issue&itemId=38249253)
- Certificate Backlog [Item 8 - Stay logged in](https://github.com/orgs/Educado-App/projects/5/views/1?pane=issue&itemId=40256182)

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Notes for Reviewers

Security concerns:
- Passwords should have more strict requirements
- HTTPS should be used in the future

### CODE REVIEW

File: eml/screens/errors/errorHandler.js
Comment (Mafusn) [inline eml/screens/errors/errorHandler.js:1]: Are the three if statements with the variable 'something' inside supposed to be in the dev branch?

File: eml/test/components/sum.js
Comment (Mafusn) [inline eml/test/components/sum.js:1]: Maybe just delete this file?

**Review by Mafusn — CHANGES_REQUESTED** [review]
Only got two comments

**Review by Havkost — CHANGES_REQUESTED** [review]
## Nice job
Would be nice to implement Montserrat as the default font before we merge into dev.
Super great work with the testing! :)

File: eml/test/components/sum.js
Comment (Jatewo) [reply eml/test/components/sum.js:1]: Should already be deleted :)

**Review by Havkost — APPROVED** [review]
## Looks good, approving
We could still add a bit more testing for emails, but we need to agree 100% on the correct form of emails first. The regex file in assets is a good first step 👍

**Review by TheMOHD3886 — COMMENTED** [review]
ye

**Pull request merged**